### PR TITLE
update Siesta EasyBlock to support GCC 10+ by adding -fallow-argument-mismatch Fortran compiler option

### PR DESCRIPTION
**Pull request by SebastianAchilles**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by SebastianAchilles** [comment]
Test report by @SebastianAchilles
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Siesta-4.1.5-foss-2020a.eb_ 
 * **SUCCESS** _Siesta-4.1-MaX-1.0-intel-2019b.eb_ 
 * **SUCCESS** _Siesta-4.1-MaX-1.0-intel-2019b-PEXSI.eb_ 

Build succeeded for 3 out of 3 (3 easyconfigs in total)
cnx1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/1469ea616dadb63a6d481c403d218198 for a full test report.

File: easybuild/easyblocks/s/siesta.py [hunk 1]
```diff
@@ -108,6 +108,12 @@ def configure_step(self):
             (r"^(FPPFLAGS_free_F90\s*=.*)$", r"\1 -ffree-line-length-none"),
         ]
 
+        gfortran_flags = ''
+        gcc_version = get_software_version('GCCcore') or get_software_version('GCC')
+        if LooseVersion(gcc_version) >= LooseVersion('10.0'):
+            # -fallow-argument-mismatch is required when compiling with GCC 10.x & more recent
+            gfortran_flags = '-fallow-argument-mismatch'
```
Comment (boegel) [inline easybuild/easyblocks/s/siesta.py:115]: Hmm, not sure this is a good approach...

Surely Siesta will be updated such that `-fallow-argument-mismatch` is no longer needed, so this really belongs in the easyconfig no?

Comment (SebastianAchilles) [reply easybuild/easyblocks/s/siesta.py:115]: Currently the Siesta EasyBlock does not allow to set this parameter in the EasyConfig (at least I did not manage). That is why i have added it to the EasyBlock.

A similar approach is used in
- BerkeleyGW https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/b/berkeleygw.py#L122
- CP2K https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/c/cp2k.py#L551-L559
- QuantumESPRESSO https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/q/quantumespresso.py#L184-L185

Comment (boegel) [reply easybuild/easyblocks/s/siesta.py:115]: Ah, ok, forgot about that... In that case, my complaint is moot :)

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _wxPython-4.0.4-GCC-8.2.0-2.31.1-Python-3.7.2.eb_ 
 * **FAIL (build issue)** _wxPython-3.0.2.0-foss-2018b-Python-2.7.15.eb_ (partial log available at https://gist.github.com/1daf1ce58efd2fc3a341fb5881d0a3ba)

Build succeeded for 1 out of 2 (2 easyconfigs in total)
node3120.skitty.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/2d573bfad02b0b235cbff6d39b3f96f1 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Siesta-4.1-MaX-1.0-intel-2019b-PEXSI.eb_ 
 * **SUCCESS** _Siesta-4.1-MaX-1.0-intel-2019b.eb_ 
 * **SUCCESS** _Siesta-4.1.5-foss-2020a.eb_ 

Build succeeded for 3 out of 3 (3 easyconfigs in total)
node3120.skitty.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/1a2cf805029ac4f83ba7b164989b6c2b for a full test report.

File: easybuild/easyblocks/s/siesta.py (hunk 1 above)
Comment (lexming) [reply easybuild/easyblocks/s/siesta.py:115]: I agree to set this in the easyblock but this needs to be constrained not only by version of compiler, but also by version of SIESTA. If the fututure version with the fix is still not known, we should set an upper limit to the current major version which will force us to not forget about this setting and review it in the future.

Comment (boegel) [reply easybuild/easyblocks/s/siesta.py:115]: @SebastianAchilles Let's include @lexming's suggestion?

Comment (smoors) [reply easybuild/easyblocks/s/siesta.py:115]: @SebastianAchilles maybe also report this upstream?

Comment (akesandgren) [reply easybuild/easyblocks/s/siesta.py:115]: @SebastianAchilles ping regarding including lexming's suggestions

Comment (boegel) [reply easybuild/easyblocks/s/siesta.py:115]: fixed in ec82d64

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Siesta-4.1.5-intel-2020a.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2604.swalot.os - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/bdcb772d41cdd258bebfedbd1310de6a for a full test report.

**Pull request merged**